rust/src/types/amount.rs

```diff
@@ -1,3 +1,4 @@
+#![allow(clippy::multiple_inherent_impl)]
```

Comment:
We already talked about this, the order of declaring the structs. 
Now that clippy is on my side, I am going to ask you to place together member implementations. 

I am not a fan of writing extra lines to suppress extra tooling.